### ISSUE TITLE
PodDisruptionBudget: deprecate policy/v1beta1, use policy/v1 instead

### ISSUE THREAD

**Issue by CompuGlobalHyperMegaNet-VP** [hashicorp/vault-helm#667]
**Is your feature request related to a problem? Please describe.**
from https://kubernetes.io/docs/reference/using-api/deprecation-guide/#poddisruptionbudget-v125:
_The policy/v1beta1 API version of PodDisruptionBudget will no longer be served in v1.25.
Migrate manifests and API clients to use the policy/v1 API version, available since v1.21._

**Describe the solution you'd like**
also from https://kubernetes.io/docs/reference/using-api/deprecation-guide/#poddisruptionbudget-v125:
_All existing persisted objects are accessible via the new API_
Therefore, something like this 
```
{{ if $.Capabilities.APIVersions.Has "policy/v1/PodDisruptionBudget" -}}
apiVersion: policy/v1
{{- else -}}
apiVersion: policy/v1beta1
{{- end }}
kind: PodDisruptionBudget
```
should work for templates\server-disruptionbudget.yaml

**Comment by tvoran** [comment]
Thanks for raising this! It's definitely something we should start supporting soon, and it also applies to the [injector](https://github.com/hashicorp/vault-helm/blob/609444d9d9d1c9c569f0edb6b2b3d8f8e60ef6dc/templates/injector-disruptionbudget.yaml). PRs welcome 😄 

Fortunately `spec.selector` is not empty in either case, so I don't think we'll need to worry about the changes in **policy/v1** detailed here: https://kubernetes.io/docs/reference/using-api/deprecation-guide/#poddisruptionbudget-v125

**Comment by hilyas** [comment]
Yep, this is needed.

**Comment by sharkannon** [comment]
@tvoran @hilyas https://github.com/hashicorp/vault-helm/pull/710 for review

**Comment by tvoran** [comment]
Thanks @sharkannon! Fixed in #710.
